### PR TITLE
Fix client-side preamble routing.

### PR DESCRIPTION
**Pull request by jmcarp**
Fix back/forward navigation for preamble sections. To verify, navigate to a series of preamble and cfr change sections, then use the back and forward buttons. Nothing should blow up.

### CODE REVIEW

File: regulations/static/regulations/js/source/router.js [hunk 1]
```diff
@@ -23,35 +23,31 @@ else {
...
-            // ask the view not to route, its not needed
-            options.noRoute = true;
-
-            MainEvents.trigger('section:open', section, options, 'reg-section');
+          this.openSection(section, 'reg-section');
         },
 
         loadPreamble: function(docId, section) {
-            var parts = [docId];
-            if (section) {
```
Comment (xtine) [inline regulations/static/regulations/js/source/router.js]: Was it possible before that there wouldn't be `section` available when loading the preamble?

File: regulations/static/regulations/js/source/router.js [hunk 2]
```diff
@@ -23,35 +23,31 @@ else {
...
-              noRoute: true
-            };
+        loadCfrChanges: function(docId, section) {
+          var parts = [docId, 'cfr', section];
+          this.openSection(parts.join('-'), 'preamble-section');
+        },
 
-            MainEvents.trigger('section:open', section, options, 'preamble-section');
+        openSection: function(id, type) {
+          var options = {
```
Comment (xtine) [inline regulations/static/regulations/js/source/router.js]: 👍 on setting options in a separate function.

File: regulations/static/regulations/js/source/router.js (hunk 1 above)
Comment (jmcarp) [inline regulations/static/regulations/js/source/router.js]: Yep--previously you could load a url like `/preamble/2016_02749`, but those urls now redirect to the 0th preamble section, like `/preamble/2016_02749/intro`.

**Comment by cmc333333** [comment]
:+1: works locally

**Pull request merged**